src/gei/AdoApiFactory.cs

```diff
@@ -17,11 +19,17 @@ public AdoApiFactory(OctoLogger octoLogger, HttpClient client, EnvironmentVariab
             _versionProvider = versionProvider;
         }
 
-        public virtual AdoApi Create(string personalAccessToken = null)
+        public virtual AdoApi Create()
+        {
+            return Create(null, null);
+        }
+
+        public virtual AdoApi Create(string adoServerUrl, string personalAccessToken)
```

Comment:
I deliberately got rid of the optional args. I personally think the overrides is clearer, but also with 2 optional strings its way to easy to do factory.Create(adoPat) not realizing that you actually passed a PAT into the URL field.

Reply:
The only problem with this is as you said it would be impossible to only provide `personalAccessToken` and we need to provide both now. So we will now end up with lots of calls like following which I personally don't like:
```c#
var adoApi = adoApiFactory.Create(null, "PAT");
```

Reply:
what do you suggest?

Reply:
There is one other option too which is to set a default value for `adoServerUrl` but then we need to put the default urls (for both GitHub and Ado) into one place (i.e. inside `Defaults.cs`)

Reply:
My ultimate suggestion is to leave the default values as is and just make the `Create()` internal since it's only going to be used in unit tests anyway.

Reply:
I made it real simple, 2 args, neither optional. In our code we're pretty much always passing both anyways, and now the test mocks match thee signature of the code under test.